### PR TITLE
feat: add storage_mode option in create table sql

### LINKED ISSUE #1294: add `storage_mode` option for `create table` statement
**Is your feature request related to a problem? Please describe.**
disk-based storage

**Describe the solution you'd like**
OPTIONS(storage_mode="HDD")

### PR DESCRIPTION
**Pull request by zhanghaohit**
close #1294

### CODE REVIEW

File: hybridse/include/node/node_manager.h [hunk 1]
```diff
@@ -309,6 +309,7 @@ class NodeManager {
     CreatePlanNode *MakeCreateTablePlanNode(
         const std::string& db_name,
         const std::string &table_name, int replica_num, int partition_num,
+        const std::string& storage_mode,
```
Comment (aceforeverd) [inline hybridse/include/node/node_manager.h]: can this be a enum ?

Comment (zhanghaohit) [reply hybridse/include/node/node_manager.h]: Actually there is an `enum StorageMode` in `src/proto/common.proto`. But I'm not sure it is good to include header `src/proto/common.pb.h` in `hybridse` codebase. 

Is it ok to add `${PROJECT_SOURCE_DIR}/../src/proto` or `${PROJECT_SOURCE_DIR}/..`  to `include_directories` in `hybridse/CMakeLists.txt`?

Comment (aceforeverd) [reply hybridse/include/node/node_manager.h]: better not, currently hybridse is a dependency to src. Maybe we can define separate enum, save space at least

File: hybridse/include/node/sql_node.h [hunk 2]
```diff
@@ -1795,19 +1820,43 @@ class InsertStmt : public SqlNode {
     const std::vector<ExprNode *> values_;
     const bool is_all_;
 };
+
+class StorageModeNode : public SqlNode {
```
Comment (dl239) [inline hybridse/include/node/sql_node.h:1824]: it no need to add such node after #1546

**Comment by zhanghaohit** [comment]
[long time window - storage engine](https://trello.com/c/UxAMvWh5/5-long-time-window-storage-engine)

**Comment by dl239** [comment]
the storage_mode usage in sdk will be in another pr?

**Comment by zhanghaohit** [comment]
> the storage_mode usage in sdk will be in another pr?

Yes. Will create another PR after this pr and  #1483 are merged

**Review by dl239 — APPROVED** [review]

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**